src/content/docs/concepts/size.mdx

```diff
@@ -10,3 +10,91 @@ sidebar:
+- [codegen-units:](https://doc.rust-lang.org/cargo/reference/profiles.html#codegen-units) Faster compile times but slower running code.
+  - `dev` `omit | 256`: Enabling incremental builds sets this to 256 by default.
+  - `release` `1`: Reducing compile time isn't interesting for release builds when it reduces performance.
+- [lto:](https://doc.rust-lang.org/cargo/reference/profiles.html#lto) Link time optimizations.
+  - `dev` `omit | "off"`: Increases compile time for the sake of performance. The default is false which may still perform limited optimizations.
+  - `release` `true`: Higher performance for release builds.
+- [opt-level:](https://doc.rust-lang.org/cargo/reference/profiles.html#opt-level) Determines the focus of the compiler.
+  - `dev` `omit | 1`: Either omit it or increase it in order to enhance the speed of the IPC during development.
+  - `release` `"s"`: This level de-prioritizes performance for the sake of reducing binary size without disregarding it completely.
+- [panic:](https://doc.rust-lang.org/cargo/reference/profiles.html#panic) Reduce size by removing panic unwinding.
+  - `dev` `omit | "unwind"`: By default unwinding is used, which is preferable during development for debugging.
+  - `release` `"abort"`: Increases performance and reduces size by removing panic handlers you're likely not using in production anyway.
+- [strip:](https://doc.rust-lang.org/cargo/reference/profiles.html#strip) Strip either symbols or debuginfo from a binary.
+  - `dev` `omit | false`: You want debug symbols during development.
+  - `release` `true`: Reduces app size by removing debug symbols.
+- [rpath:](https://doc.rust-lang.org/cargo/reference/profiles.html#rpath) Assists in finding the dynamic libraries the binary requires by hard coding information into the binary.
+  - `dev` `omit | false`: Irrelevant during development.
+  - `release` `omit | false`: Reduces app size, the default is false.
+
+### Nightly Features
+
+- [trim-paths:](https://rust-lang.github.io/rfcs/3127-trim-paths.html) Removes potentially privileged information from binaries.
+  - `dev` `omit`: Irrelevant during development.
+  - `release` `"all"`: Removes paths from the built binary which may in some cases be seen as privileged data.
+- [rustflags:](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#profile-rustflags-option) Sets Rust compiler flags on a profile by profile basis.
+  - `"-Cdebuginfo"`: Whether debuginfo symbols should be included in the build.
+    - `dev` `omit`: You want debuginfo during development.
+    - `release` `0`: Depending on your Rust version this may or may not have an effect on the symbols that get stripped.
+  - `"-Zthreads"`: Increases the number of threads used during compilation.
+    - `dev` `8`: Adjust as is appropriate based on developer and CI/CD hardware.
+    - `release` `8`: Has no effect on the resulting binary beyond it compiling faster.
```

Comment:
Tbh, I don't like this section at all, it seems bloated and also needs to understand a special syntax.

I think I'd prefer the simplicity of the snippet from v1 docs which just have a small explanation next to each option

```toml
[profile.release]
panic = "abort" # Strip expensive panic clean-up logic
codegen-units = 1 # Compile crates one after another so the compiler can optimize better
lto = true # Enables link to optimizations
opt-level = "s" # Optimize for binary size
strip = true # Remove debug symbols
```

We can then replace this bloated section with just links to the official cargo docs about it:
```md

# References
- [lto](link-to-cargo-docs)
..etc
```

Reply:
Restructured to what I think is closer to what you want.